src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -395,68 +411,63 @@ private boolean isDatabaseReadyForBackup(BibDatabaseContext context) {
      * <p>
      * Example: *jabref-authors.bib – testbib
      */
-    public void updateTabTitle(boolean isChanged) {
-        boolean isAutosaveEnabled = preferences.getLibraryPreferences().shouldAutoSave();
-
-        DatabaseLocation databaseLocation = bibDatabaseContext.getLocation();
-        Optional<Path> file = bibDatabaseContext.getDatabasePath();
 
-        StringBuilder tabTitle = new StringBuilder();
-        StringBuilder toolTipText = new StringBuilder();
+    public void updateTabTitle(boolean isModified) {
+        // initialize string joiner separate different elements by braces
+        StringJoiner tabTitle = new StringJoiner(" ");
 
-        if (file.isPresent()) {
-            // Modification asterisk
-            if (isChanged && !isAutosaveEnabled) {
-                tabTitle.append('*');
-            }
-
-            // Filename
-            Path databasePath = file.get();
-            String fileName = databasePath.getFileName().toString();
-            tabTitle.append(fileName);
-            toolTipText.append(databasePath.toAbsolutePath());
-
-            if (databaseLocation == DatabaseLocation.SHARED) {
-                tabTitle.append(" \u2013 ");
-                addSharedDbInformation(tabTitle, bibDatabaseContext);
-                toolTipText.append(' ');
-                addSharedDbInformation(toolTipText, bibDatabaseContext);
+        try {
+            if (bibDatabaseContext.getDatabasePath().isPresent()) {
+                // get the database path
+                Path databasePath = bibDatabaseContext.getDatabasePath().get();
+                // Try to get Git status if available
+                addGitStatusToTitle(tabTitle);
+                // Regular tab title with file name
+                String fileNamePart = databasePath.getFileName().toString();
+                tabTitle.add(fileNamePart);
+            } else {
+                tabTitle.add(Localization.lang("untitled"));
             }
+        } catch (Exception e) {
+            tabTitle.add(Localization.lang("untitled"));
+        }
```

Comment:
Do not use Exceptions for control flow, but only for exceptional states.
https://github.com/HugoMatilla/Effective-JAVA-Summary?tab=readme-ov-file#57-use-exceptions-only-for-exceptional-conditions

Do not catch generic Exceptions, but standard exceptions according to the abstraction level.
https://github.com/HugoMatilla/Effective-JAVA-Summary?tab=readme-ov-file#60-favor-the-use-of-standard-exceptions